test/diags_test.py

```diff
@@ -106,7 +106,10 @@ def execute(self, test_str, imagefilename, imagethreshold, ncfiles, rtol, atol):
         proc = subprocess.Popen([self.diagstr], shell=True)
         proc_status = proc.wait()
         if proc_status!=0: 
-            raise DiagError("diags run failed")
+            proc = subprocess.Popen([self.diagstr], shell=True)
+            proc_status = proc.wait()
+            if proc_status!=0: 
```

Comment:
Why retry on failure?   Popen sometimes fails randomly.  It reflects a problem with the test scripts, not the code being tested.  The retry is only a workaround.  We should figure out the causes of the failures and fix them if possible.